torch/futures/__init__.py

```diff
@@ -23,6 +23,25 @@ class Future(torch._C.Future, Generic[T], metaclass=_PyFutureMeta):
     also exposes a set of APIs to add callback functions and set results.
     """
 
+    def __init__(self, devices: Optional[List[Union[int, str, torch.device]]] = None):
+        r"""
+        Create a ``Future`` that is allowed to use (a subset of) the given
+        devices. All devices must be CUDA devices (if PyTorch supports CUDA).
```

Comment:
Curious, why do we need "if PyTorch supports CUDA" here?

Reply:
I meant this as "if you're on a CPU-only build you cannot pass CUDA devices here". In other words, if the user is passing in a CUDA device, but we cannot create a CUDAFuture (because that code wasn't built!), then we'll throw an error.

Happy to rephrase this if you have suggestions!

Reply:
Maybe "if `torch.cuda.is_available()` returns `True`"?